File: src/content/docs/topics/cs/leetcode-150/binary-search/704-binary-search.py
def search(nums: list, target: int) -> int:
    pass  # TODO: implement

def _run_tests():
    assert search([-1, 0, 3, 5, 9, 12], 9) == 4
    assert search([-1, 0, 3, 5, 9, 12], 2) == -1
    assert search([5], 5) == 0
    assert search([5], 3) == -1
    assert search([-1, 0, 3, 5, 9, 12], -1) == 0
    assert search([-1, 0, 3, 5, 9, 12], 12) == 5
    # --- large-input timing ---
    import time as _t
    _arr = list(range(100000))
    _t0 = _t.perf_counter()
    search(_arr, 99999)
    _ms = (_t.perf_counter() - _t0) * 1000
    print(f'perf binary search n=100000: {_ms:.1f}ms')
    print('all tests pass')

if __name__ == '__main__':
    _run_tests()

File: tests/fixtures/swift-coding-problem-contract/704-binary-search.swift
// LEETCODE_TYPE: Solution

func expectEqual<T: Equatable>(
    _ actual: T,
    _ expected: T,
    _ message: String = "",
    file: StaticString = #fileID,
    line: UInt = #line
) {
    guard actual == expected else {
        let detail = message.isEmpty ? "values differ" : message
        fatalError("\(file):\(line): \(detail). Expected \(expected), got \(actual)")
    }
}

func expectTrue(
    _ condition: @autoclosure () -> Bool,
    _ message: String = "expected true",
    file: StaticString = #fileID,
    line: UInt = #line
) {
    guard condition() else {
        fatalError("\(file):\(line): \(message)")
    }
}

func reportSuccess() {
    print("All Swift tests passed")
}

final class Solution {
    func search(_ nums: [Int], _ target: Int) -> Int {
        // TODO: Implement
        fatalError("TODO: Implement")
    }
}

func runTests() {
    let solution = Solution()
    expectEqual(solution.search([-1, 0, 3, 5, 9, 12], 9), 4)
    expectEqual(solution.search([-1, 0, 3, 5, 9, 12], 2), -1)
    expectTrue(solution.search([5], 5) == 0)
    reportSuccess()
}

runTests()
